RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -984,11 +988,17 @@ void MuonIdProducer::fillArbitrationInfo( reco::MuonCollection* pOutputMuons )
      // chamberIter1
      for ( auto& chamber1 : muon1.matches() )
      {
-       if(chamber1.segmentMatches.empty()) continue;
+       // segmentIter1
+       std::vector<reco::MuonSegmentMatch> * segmentMatches1 = nullptr;
+       if      (muonType == reco::Muon::TrackerMuon) segmentMatches1 =  & chamber1.segmentMatches;
```

Comment:
the `if`s are all on the same quantity, so only the first one can throw (the code will never get to the others if the first one does).
Anyway, I have refactored it into a helper of `MuonIdProducer` (alternatively it could go in `MuonChamberMatch`)